lambdas/notification-configuration/index.py

```diff
@@ -0,0 +1,66 @@
+"""
+Set up indexer notification on data bucket.
+
+Remove notification on delete.
+"""
+import boto3
+
+from t4_lambda_shared.cfnresponse import send, SUCCESS, FAILED
+
+def create_mappings(params):
+    """ Sets up desired mappings after checking no mappings currently exist. """
+    s3 = boto3.client('s3')
+    existing = s3.get_bucket_notification_configuration(Bucket=params['Bucket'])
+    if 'TopicConfigurations' in existing \
+            or 'QueueConfigurations' in existing \
+            or 'LambdaFunctionConfigurations' in existing:
+        raise Exception('Unable to attach notifications. Notification already configured '
+                'on bucket {}.'.format(params['Bucket']))
+
+    s3.put_bucket_notification_configuration(**params)
+    return
+
+def handler(event, context):
+    """ Top-level handler for custom resource """
+    print('Changing bucket notification settings')
+    try:
+        params = dict(event['ResourceProperties'])
+        del params['ServiceToken']
```

Comment:
Is it better to black list the `ServiceToken` field or whitelist all other fields? Can you add a comment to that effect? Seems safer to explicitly copy fields.